From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Sing-Han Chen <singhanc@nvidia.com>
Date: Thu, 20 Jun 2024 06:11:39 +0000
Subject: [PATCH 158/245] NVIDIA: SAUCE: phy: xusb: skip USB2 PAD tracking and
 fuse

The USB2 PAD tracking clock is not necessary a dedicated clock.
And fuse settings is not ready for everyone in this stage.

http://nvbugs/4703456

Change-Id: I2c49e269a76e468e1bedb2ab24e4def125c5d3a0
Signed-off-by: Sing-Han Chen <singhanc@nvidia.com>
(cherry picked from commit 1de24d5bcccb00131b9aa189e4a340be821ec64b)
Signed-off-by: Bodla Rakesh Babu <rbodla@nvidia.com>

diff --git a/drivers/phy/tegra/xusb-tegra186.c b/drivers/phy/tegra/xusb-tegra186.c
index 32186e0af611e..0fdf5dd83f9c8 100644
--- a/drivers/phy/tegra/xusb-tegra186.c
+++ b/drivers/phy/tegra/xusb-tegra186.c
@@ -955,8 +955,13 @@ static int tegra186_utmi_phy_power_on(struct phy *phy)
 	value = padctl_readl(padctl, XUSB_PADCTL_USB2_OTG_PADX_CTL0(index));
 	value &= ~USB2_OTG_PD_ZI;
 	value |= TERM_SEL;
-	value &= ~HS_CURR_LEVEL(~0);
 
+	if (padctl->soc->ignore_fuse) {
+		padctl_writel(padctl, value, XUSB_PADCTL_USB2_OTG_PADX_CTL0(index));
+		goto ignore_fuse;
+	}
+
+	value &= ~HS_CURR_LEVEL(~0);
 	if (!padctl->is_xhci_iov) {
 		if (usb2->hs_curr_level_offset) {
 			int hs_current_level;
@@ -984,6 +989,7 @@ static int tegra186_utmi_phy_power_on(struct phy *phy)
 	value |= RPD_CTRL(priv->calib.rpd_ctrl);
 	padctl_writel(padctl, value, XUSB_PADCTL_USB2_OTG_PADX_CTL1(index));
 
+ignore_fuse:
 	tegra186_utmi_pad_power_on(phy);
 
 	return 0;
@@ -1095,7 +1101,7 @@ tegra186_usb2_pad_probe(struct tegra_xusb_padctl *padctl,
 	}
 
 	if (!padctl->is_xhci_iov) {
-		priv->usb2_trk_clk = devm_clk_get(&pad->dev, "trk");
+		priv->usb2_trk_clk = devm_clk_get_optional(&pad->dev, "trk");
 		if (IS_ERR(priv->usb2_trk_clk)) {
 			err = PTR_ERR(priv->usb2_trk_clk);
 			dev_dbg(&pad->dev, "failed to get usb2 trk clock: %d\n", err);
@@ -1597,7 +1603,7 @@ tegra186_xusb_padctl_probe(struct device *dev,
 	if (IS_ERR(priv->ao_regs))
 		return ERR_CAST(priv->ao_regs);
 
-	if (!is_xhci_iov) {
+	if (!is_xhci_iov && !soc->ignore_fuse) {
 		err = tegra186_xusb_read_fuse_calibration(priv);
 		if (err < 0)
 			return ERR_PTR(err);
@@ -1838,6 +1844,7 @@ const struct tegra_xusb_padctl_soc tegra264_xusb_padctl_soc = {
 	.poll_trk_completed = true,
 	.trk_hw_mode = true,
 	.supports_lp_cfg_en = true,
+	.ignore_fuse = true,
 };
 EXPORT_SYMBOL_GPL(tegra264_xusb_padctl_soc);
 #endif
diff --git a/drivers/phy/tegra/xusb.h b/drivers/phy/tegra/xusb.h
index 09cb98c273ce6..2b4a5f8271e1d 100644
--- a/drivers/phy/tegra/xusb.h
+++ b/drivers/phy/tegra/xusb.h
@@ -438,6 +438,7 @@ struct tegra_xusb_padctl_soc {
 	bool trk_update_on_idle;
 	bool supports_lp_cfg_en;
 	bool has_per_pad_term;
+	bool ignore_fuse;
 };
 
 struct tegra_xusb_padctl {
-- 
2.43.0

